py/core/main/api/v3/collections_router.py

```diff
@@ -0,0 +1,782 @@
+            The user must have appropriate permissions to modify the collection.
+            """
+            if (
+                not auth_user.is_superuser
+                and id not in auth_user.collection_ids
+            ):
+                raise R2RException(
+                    "The currently authenticated user does not have access to the specified collection.",
+                    403,
+                )
+
+            return await self.services["management"].update_collection(  # type: ignore
+                id, name=config.name, description=config.description
+            )
+
+        @self.router.delete(
+            "/collections/{id}",
+            summary="Delete collection",
+            openapi_extra={
+                "x-codeSamples": [
+                    {
+                        "lang": "Python",
+                        "source": textwrap.dedent(
+                            """
+                            from r2r import R2RClient
+
+                            client = R2RClient("http://localhost:7272")
+                            # when using auth, do client.login(...)
+
+                            result = client.collections.delete("123e4567-e89b-12d3-a456-426614174000")
+                        """
+                        ),
+                    },
+                    {
+                        "lang": "cURL",
+                        "source": textwrap.dedent(
+                            """
+                            curl -X DELETE "https://api.example.com/v3/collections/123e4567-e89b-12d3-a456-426614174000" \\
+                                 -H "Authorization: Bearer YOUR_API_KEY"
+                        """
+                        ),
+                    },
+                ]
+            },
+        )
+        @self.base_endpoint
+        async def delete_collection(
+            id: UUID = Path(
+                ...,
+                description="The unique identifier of the collection to delete",
+            ),
+            auth_user=Depends(self.providers.auth.auth_wrapper),
+        ) -> ResultsWrapper[bool]:
+            """
+            Delete an existing collection.
+
+            This endpoint allows deletion of a collection identified by its UUID.
+            The user must have appropriate permissions to delete the collection.
+            Deleting a collection removes all associations but does not delete the documents within it.
+            """
+            if (
+                not auth_user.is_superuser
+                and id not in auth_user.collection_ids
+            ):
+                raise R2RException(
+                    "The currently authenticated user does not have access to the specified collection.",
+                    403,
+                )
+
+            await self.services["management"].delete_collection(id)
+            return True  # type: ignore
+
+        @self.router.post(
+            "/collections/{id}/documents/{document_id}",
```

Comment:
This should really be a PUT request It seems more appropriate here since:
- We're establishing a specific relationship at a known URL path
- The operation is idempotent—adding the same document to a collection multiple times should have the same result
- The URL clearly represents the resource relationship being established

A POST would be more appropriate if you were creating new documents rather than adding existing ones.